p2p/simulations/network.go

```diff
@@ -293,7 +293,9 @@ func (self *Network) Disconnect(oneID, otherID discover.NodeID) error {
 
 // DidConnect tracks the fact that the "one" node connected to the "other" node
 func (self *Network) DidConnect(one, other discover.NodeID) error {
-	conn, err := self.GetOrCreateConn(one, other)
+	self.lock.Lock()
+	defer self.lock.Unlock()
+	conn, err := self.getOrCreateConn(one, other)
```

Comment:
Yep, great find, I recently ran a `go test -race` which reported this race condition, but forgot to put that in as I was debugging something else. Thanks for that